### PR TITLE
[ISSUE 182]Improve instance health flag

### PR DESCRIPTION
**Pull request by universefeeler**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

support service health flag show 

## Brief changelog

add triggerFlag for service 

## Verifying this change

Ok

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by nkorange** [comment]
Related issue is #182

File: naming/src/main/java/com/alibaba/nacos/naming/controllers/CatalogController.java [hunk 1]
```diff
@@ -284,18 +284,8 @@ private JSONObject serviceList(HttpServletRequest request) {
...
-                if (instance.isHealthy()) {
-                    validCount++;
-                }
-
-            }
-
-            serviceView.setHealthyInstanceCount(validCount);
-
+            serviceView.setHealthyInstanceCount(service.healthyInstanceCount());
+            serviceView.setTriggerFlag(service.isTriggerFlag()?"True":"False");
```
Comment (nkorange) [inline naming/src/main/java/com/alibaba/nacos/naming/controllers/CatalogController.java]: 为什么True首字母大写？

File: api/src/main/java/com/alibaba/nacos/api/naming/pojo/Service.java [hunk 2]
```diff
@@ -102,10 +107,19 @@ public void addMetadata(String key, String value) {
         this.metadata.put(key, value);
     }
 
+    public boolean isTriggerFlag() {
```
Comment (nkorange) [inline api/src/main/java/com/alibaba/nacos/api/naming/pojo/Service.java]: 该字段其实不需要存储，只需要在查询实例列表的时候返回就行了。

File: naming/src/main/java/com/alibaba/nacos/naming/controllers/CatalogController.java (hunk 1 above)
Comment (universefeeler) [reply naming/src/main/java/com/alibaba/nacos/naming/controllers/CatalogController.java]: 没有其他原因，在前端显示感觉整齐些

File: api/src/main/java/com/alibaba/nacos/api/naming/pojo/Service.java (hunk 2 above)
Comment (universefeeler) [reply api/src/main/java/com/alibaba/nacos/api/naming/pojo/Service.java]: 期初我也是这么想的。1， 考虑到每次查询都需要计算 2，可能有其他api也会用到该值 所以计算好存储了

File: naming/src/main/java/com/alibaba/nacos/naming/controllers/CatalogController.java (hunk 1 above)
Comment (universefeeler) [reply naming/src/main/java/com/alibaba/nacos/naming/controllers/CatalogController.java]: 这个我改为全小写吧，与boolean的值保持一致

File: api/src/main/java/com/alibaba/nacos/api/naming/pojo/Service.java (hunk 2 above)
Comment (nkorange) [reply api/src/main/java/com/alibaba/nacos/api/naming/pojo/Service.java]: 只需要在/instance/list这接口里设置一下就行了，这个接口之前就有计算逻辑的。

Comment (universefeeler) [reply api/src/main/java/com/alibaba/nacos/api/naming/pojo/Service.java]: 嗯，好

File: naming/src/main/java/com/alibaba/nacos/naming/core/Service.java [hunk 3]
```diff
@@ -191,8 +191,16 @@ public int healthyInstanceCount() {
...
     }
 
-    public boolean meetProtectThreshold() {
-        return (healthyInstanceCount() * 1.0 / allIPs().size()) <= getProtectThreshold();
+    public boolean triggerFlag() {
+        if(0 == healthyInstanceCount()) {
+            return true;
+        }
+        Float compareValue = 0.0F;
+        if(Float.floatToIntBits(compareValue) == getProtectThreshold()) {
```
Comment (nkorange) [inline naming/src/main/java/com/alibaba/nacos/naming/core/Service.java]: 保护阈值为0，也可能触发保护的，也就是所有实例都不健康的时候

File: console/src/main/resources/static/console-fe/src/locales/zh-CN.js [hunk 4]
```diff
@@ -101,6 +101,7 @@ const I18N_CONF = {
     columnClusterCount: '集群数目',
     columnIpCount: '实例数',
     columnHealthyInstanceCount: '健康实例数',
+    columnTriggerFlag: '触发阈值',
```
Comment (nkorange) [inline console/src/main/resources/static/console-fe/src/locales/zh-CN.js]: 应该叫“触发雪崩保护”？我也不确定哪个名字好一点

File: console/src/main/resources/static/console-fe/src/locales/en-US.js [hunk 5]
```diff
@@ -101,6 +101,7 @@ const I18N_CONF = {
     columnClusterCount: 'Cluster Count',
     columnIpCount: 'Instance Count',
     columnHealthyInstanceCount: 'Healthy Instance Count',
+    columnTriggerFlag: 'Trigger Flag',
```
Comment (nkorange) [inline console/src/main/resources/static/console-fe/src/locales/en-US.js]: 英文和中文不对应

File: naming/src/main/java/com/alibaba/nacos/naming/core/Service.java (hunk 3 above)
Comment (universefeeler) [reply naming/src/main/java/com/alibaba/nacos/naming/core/Service.java]: 这个目前就是这样
public void updateTriggerFlag() {
        if(0 == healthyInstanceCount()) {
            setTriggerFlag(true);
            return;
        }
        Float compareValue = 0.0F;
        if(Float.floatToIntBits(compareValue) == getProtectThreshold()) {
            setTriggerFlag(false);
            return;
        }
        boolean triggerFlag = (healthyInstanceCount() * 1.0 / allIPs().size()) <= getProtectThreshold();
        setTriggerFlag(triggerFlag);
    }

File: console/src/main/resources/static/console-fe/src/locales/zh-CN.js (hunk 4 above)
Comment (universefeeler) [reply console/src/main/resources/static/console-fe/src/locales/zh-CN.js]: 因为这个保护阈值是自己设置的，叫“触发保护阈值”吧？

File: console/src/main/resources/static/console-fe/src/locales/en-US.js (hunk 5 above)
Comment (universefeeler) [reply console/src/main/resources/static/console-fe/src/locales/en-US.js]: 嗯，是的，这里我修改下

File: console/src/main/resources/static/console-fe/src/locales/zh-CN.js (hunk 4 above)
Comment (nkorange) [reply console/src/main/resources/static/console-fe/src/locales/zh-CN.js]: 触发保护阈值可以

**Pull request merged**